uwblueprint/focus-on-nature
# Feat: camp overview - adding and removing coordinators + photos

## Notion ticket link

[Camp Overview: adding/removing Camp Coordinators + Camp Counsellors](https://www.notion.so/uwblueprintexecs/Camp-Overview-adding-removing-Camp-Coordinators-Camp-Counsellors-c13e1237abee47e2b16ee5934ba6e9e7)
[Camp Overview: Adding/Removing Camp Volunteers](https://www.notion.so/uwblueprintexecs/Camp-Overview-Adding-Removing-Camp-Volunteers-b3889cf9a74e45328e06a44362dac4db)
[Camp Overview: Camp Photos](https://www.notion.so/uwblueprintexecs/Camp-Overview-Camp-Photos-3c56402473f646c4a6b985a9925969ea)


## Implementation description
- fetching the camp corresponding to the id specified in the url (/camp/:id)
- displaying camp photo
- ability to update coordinators/counsellors/volunteers
- changed type of `volunteer` to string from string[] 
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/40839109/187057066-604b5017-09a0-4f38-aecd-a1f35595e762.png">
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/40839109/187057139-6421d610-bd46-49d0-bf89-b771f6489c7a.png">
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/40839109/187057073-c94b3829-b66f-4ffa-bb72-8bf74a908812.png">

Next steps:
- integrate the top half of the camp overview page with the bottom half which was merged in earlier
- camp statuses


## Steps to test
1. Build containers and go to http://localhost:3000/admin/camp/{id} (make sure you login as an admin to the tool) 
2. Test updating camp counsellors, coordinators, volunteers
3. Check that the changes are reflected in the database after navigating away from the select fields
4. Check that the camp photo displays correctly. 

A lot of the camps in our db have missing/incorrectly formatted fields, use the following links to test:
- with camp photo: http://localhost:3000/admin/camp/630a63450eb59852f9d01b1e
- without camp photo: http://localhost:3000/admin/camp/630ae35648363c600c53b42e


## What should reviewers focus on?
* Code quality
* Any way to add dividers/separators between menu items in the select dropdown? I couldn't figure it out :/
* Any other styling issues that impact usability

## Checklist
